Sources/TIM/AppAuth/AppAuthController.swift

```diff
@@ -151,18 +151,22 @@ final class AppAuthController {
                     clientID: self.credentials.clientId,
                     clientSecret: nil,
                     scopes: self.credentials.scopes,
-                    refreshToken: refreshToken,
+                    refreshToken: refreshToken.token,
                     codeVerifier: nil,
                     additionalParameters: nil
                 )
                 OIDAuthorizationService.perform(request) { [weak self] (token: OIDTokenResponse?, error: Error?) in
                     let result: Result<JWT, TIMAuthError>
                     if let error = error {
                         result = .failure(.mapAppAuthError(error))
-                    } else if let token = token, let jwt = token.accessToken {
+                    } else if let token = token, let rawAccessToken = token.accessToken {
                         let authResponse = self?.createRestoreFakeLastAuthorizationResponse(configuration: configuration)
                         self?.authState = OIDAuthState(authorizationResponse: authResponse, tokenResponse: token, registrationResponse: nil)
-                        result = .success(jwt)
+                        if let jwt = JWT(token: rawAccessToken) {
+                            result = .success(jwt)
+                        } else {
+                            result = .failure(.failedToGetRequiredDataInToken)
```

Comment:
Det kan være lidt svært at debugge, hvis der ikke er mere kontekst med, f.eks. hvilket element(er) der mangler.

Reply:
After further investigation it turned out, that the `exp` isn't guaranteed, so it will only be the `sub` that is missing on this error.